src/dot.ts

```diff
@@ -125,20 +135,24 @@ export class Dot extends Modifier {
     return this;
   }
 
-  setDotShiftY(y) {
+  setDotShiftY(y: number): this {
     this.dot_shiftY = y;
     return this;
   }
 
-  draw() {
-    this.checkContext();
+  draw(): void {
+    const ctx = this.checkContext();
     this.setRendered();
-
-    if (!this.note || this.index === null) {
+    if (!this.note || !this.note.getStave() || this.index === undefined) {
       throw new Vex.RERR('NoAttachedNote', "Can't draw dot without a note and index.");
```

Comment:
Message "Drawing a dot requires a stave, a note, and an index".

Reply:
Done